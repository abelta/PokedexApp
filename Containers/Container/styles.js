import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'red',
    flex: 1,
    flexDirection: 'column',
  },
});

export default styles;
