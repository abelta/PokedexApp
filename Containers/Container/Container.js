import React from 'react';
import { View } from 'react-native';
import { Route, Switch } from 'react-router-native';
import { PokemonDetail, PokemonList } from '../../Containers';
import styles from './styles';

const Container = () => (
  <View style={styles.container}>
    <Switch>
      <Route path="/" component={PokemonList} exact />
      <Route path="/pokemon/:id" component={PokemonDetail} />
    </Switch>
  </View>
);

export default Container;
