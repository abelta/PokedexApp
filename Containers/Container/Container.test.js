import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { MemoryRouter } from 'react-router-native';
import Container from '.';

const middlewares = [];
const mockStore = configureStore(middlewares);
const initialState = {
  containers: {
    pokemonDetail: {
      error: false,
      loading: false,
    },
    pokemonList: {
      error: false,
      loading: false,
      offset: 0,
    },
  },
  entities: {
    pokemon: {
      pikachu: {},
    },
    pokemonIndex: {},
  },
};
const store = mockStore(initialState);

describe('Container', () => {
  describe('location is "/"', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/']} >
          <Container />
        </MemoryRouter>
      </Provider>,
    ).toJSON();

    it('renders PokemonList', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe('location is "/pokemon/:id"', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/pokemon/picachu']} >
          <Container />
        </MemoryRouter>
      </Provider>,
    ).toJSON();

    it('renders PokemonDetail', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
