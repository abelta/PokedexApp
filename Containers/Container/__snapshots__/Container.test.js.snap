// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container location is "/" renders PokemonList 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    style={undefined}
  >
    <View
      style={
        Object {
          "height": 2,
        }
      }
    />
    <RCTScrollView
      ListEmptyComponent={<Empty />}
      data={Array []}
      disableVirtualization={false}
      extraData={
        Object {
          "offset": 0,
        }
      }
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReached={[Function]}
      onEndReachedThreshold={10}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      onViewableItemsChanged={undefined}
      refreshing={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={undefined}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "borderColor": "lightgrey",
                  "borderWidth": 0.5,
                  "height": 35,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 18,
                  "textAlign": "center",
                }
              }
            >
              ...loading data
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;

exports[`Container location is "/pokemon/:id" renders PokemonDetail 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  />
</View>
`;
